packages/js/product-editor/src/components/manage-download-limits-modal/manage-download-limits-modal.tsx

```diff
@@ -0,0 +1,216 @@
+
+	function validateDownloadLimit() {
+		if ( downloadLimit && ! Number.isInteger( Number( downloadLimit ) ) ) {
+			setErrors( ( current ) => ( {
+				...current,
+				downloadLimit: __(
+					'Download limit must be an integer number',
+					'woocommerce'
+				),
+			} ) );
+			return false;
+		}
+
+		if ( Number.parseInt( downloadLimit, 10 ) < 1 ) {
+			setErrors( ( current ) => ( {
+				...current,
+				downloadLimit: sprintf(
+					__(
+						// translators: %d is the minimum value of the number input.
+						'Download limit must be greater than or equal to %d',
+						'woocommerce'
+					),
+					DOWNLOAD_LIMIT_MIN
+				),
+			} ) );
+			return false;
+		}
+
+		setErrors( ( { downloadLimit: _, ...current } ) => current );
+		return true;
+	}
+
+	function validateDownloadExpiry() {
+		if (
+			downloadExpiry &&
+			! Number.isInteger( Number( downloadExpiry ) )
+		) {
+			setErrors( ( current ) => ( {
+				...current,
+				downloadExpiry: __(
+					'Expiry period must be an integer number',
+					'woocommerce'
+				),
+			} ) );
+			return false;
+		}
+
+		if ( Number.parseInt( downloadExpiry, 10 ) < 1 ) {
+			setErrors( ( current ) => ( {
+				...current,
+				downloadExpiry: sprintf(
+					__(
+						// translators: %d is the minimum value of the number input.
+						'Expiry period must be greater than or equal to %d',
+						'woocommerce'
+					),
+					DOWNLOAD_EXPIRY_MIN
+				),
+			} ) );
+			return false;
+		}
+
+		setErrors( ( { downloadExpiry: _, ...current } ) => current );
+		return true;
+	}
+
+	const downloadLimitProps = {
+		...useNumberInputProps( {
+			value: downloadLimit,
+			onChange: setDownloadLimit,
+		} ),
+		id: useInstanceId(
+			BaseControl,
+			'product_download_limit_field'
+		) as string,
+		type: 'number',
+		className: classNames( {
+			'has-error': errors.downloadLimit,
+		} ),
+		label: __( 'Download limit', 'woocommerce' ),
+		help:
+			errors.downloadLimit ||
+			__(
+				'Decide how many times customers can download files after purchasing the product. Leave blank for unlimited re-downloads.',
+				'woocommerce'
+			),
+		placeholder: __( 'Unlimited', 'woocommerce' ),
+		suffix: (
+			<span className="woocommerce-manage-download-limits-modal__input-suffix">
+				{ __( 'times', 'woocommerce' ) }
+			</span>
+		),
+		onBlur() {
+			validateDownloadLimit();
+		},
+	};
+
+	const downloadExpiryProps = {
+		...useNumberInputProps( {
+			value: downloadExpiry,
+			onChange: setDownloadExpiry,
+		} ),
+		id: useInstanceId(
+			BaseControl,
+			'product_download_expiry_field'
+		) as string,
+		type: 'number',
+		className: classNames( {
+			'has-error': errors.downloadExpiry,
+		} ),
+		label: __( 'Expiry period', 'woocommerce' ),
+		help:
+			errors.downloadExpiry ||
+			__(
+				'Decide how long customers can access the files after purchasing the product. Leave blank for unlimited access.',
+				'woocommerce'
+			),
+		placeholder: __( 'Unlimited', 'woocommerce' ),
```

Comment:
Should this say `Never` instead?
`Unlimited` seems a little odd for a period of time ( cc: @jarekmorawski ). Only bringing this up as that is what the legacy screen also uses for placeholder.